core/storage/job/gae_models.py

```diff
@@ -143,3 +143,55 @@ class ContinuousComputationModel(base_models.BaseModel):
     # The time at which a halt signal was last sent to this batch job, in
     # milliseconds since the epoch.
     last_stopped_msec = ndb.FloatProperty(indexed=True)
+
+
+# Allowed transitions: processing --> completed --> archived
+# or processing --> failed.
+QUERY_STATUS_PROCESSING = "processing"
+QUERY_STATUS_COMPLETED = "completed"
+QUERY_STATUS_ARCHIVED = "archived"
+QUERY_STATUS_FAILED = "failed"
+
+
+# Keys for query_parameters dictionary.
+# Values corresponding to key is value specified by query submitter.
+# A query can have any combination of following parameters.
+# Valid parameters names.
+# pylint: disable=C0103
+QUERY_PARAMETERS = utils.create_enum(
+    "has_user_logged_in_since_n_days", "created_less_than_n_exploration",
+    "created_more_than_n_exploration", "edited_less_than_n_exploration",
+    "edited_more_than_n_exploration",
+    "created_or_edited_exploration_in_n_days")
+
+class QueryModel(base_models.BaseModel):
+    """Model for storing result of queries.
+
+    The id of each instance of this model is alphanumeric id of length 12
+    unique to each model instance.
+    """
+    # parameters for query given by submitter.
```

Comment:
Start each comment with a capital letter. Also, explain what the type is -- it's not clear, here.